### ISSUE TITLE
Inference with pre-trained `ms-lesion-agnostic` model

### ISSUE THREAD

**Issue by plbenveniste** [ivadomed/ms-multi-spine-challenge-2024#9]
The objective of the 1st strategy is to leverage the pre-trained model `ms-lesion agnostic` from this release https://github.com/ivadomed/ms-lesion-agnostic/releases/tag/r20250219 to segment the lesions in all contrasts. 

Here are the things I want to investigate: 
- run inference with the pretrained model on all contrasts and images (both natural images, preprocessed images and registered images) to evaluate its performance
- thresholding of the soft segmentations
- how to gather segmentations when ran on multiple contrasts
- exploring test-time-augmentation strategies

**Comment by plbenveniste** [comment]
I am running inference using the 5 folds of model 1 from the above mentionned release. 

The results are pretty encouraging. Here is an example on subject 023 for a STIR image (no segmentation available) and T2w. 

![Image](https://github.com/user-attachments/assets/545575fe-1dba-4bac-9014-96598aa0adbc)

![Image](https://github.com/user-attachments/assets/e6bbaec5-1b4f-4557-8476-731aaea7a5bd)
This displays image -> predicted soft segmentation (yellow) -> manual segmentation (red)

**Comment by plbenveniste** [comment]
I ran predictions on all types of nnUnet dataset (dataset 011 to 100) with model 1 (with `checkpoint_final.pth`). I am seeing some issues with lesions segmented outside the spinal cord such as in this case. If we decide to use this model, we should a post-processing steps which crops anything outside of a mask of the spinal cord. 
Example: 

![Image](https://github.com/user-attachments/assets/3adf78d7-011f-4131-a700-e8183366689c)

EDIT: 
For the final submission, a spinal cord segmentation mask should be applied to exclude lesion predictions located outside the cord, regardless of the chosen method.
